File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/bulk/BulkResponse.java
package fr.myprysm.vertx.elasticsearch.action.bulk;

import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;
import org.elasticsearch.rest.RestStatus;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@DataObject(generateConverter = true)
public class BulkResponse {

    private List<BulkResponseItem> items;
    private boolean errors;
    private long took;
    private RestStatus status;

    /**
     * Build a new <code>BulkResponse</code> from a <code>JsonObject</code>.
     *
     * @param json the <code>JsonObject</code>
     */
    public BulkResponse(JsonObject json) {
        BulkResponseConverter.fromJson(json, this);
    }

    /**
     * Transforms the <code>BulkResponse</code> into a <code>JsonObject</code>.
     *
     * @return the <code>JsonObject</code>
     */
    public JsonObject toJson() {
        JsonObject json = new JsonObject();
        BulkResponseConverter.toJson(this, json);
        return json;
    }
}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/bulk/BulkResponse.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.bulk

import fr.myprysm.vertx.elasticsearch.action.bulk.BulkResponse
import fr.myprysm.vertx.elasticsearch.action.bulk.BulkResponseItem
import org.elasticsearch.rest.RestStatus

fun BulkResponse(
  errors: Boolean? = null,
  items: Iterable<fr.myprysm.vertx.elasticsearch.action.bulk.BulkResponseItem>? = null,
  status: RestStatus? = null,
  took: Long? = null): BulkResponse = fr.myprysm.vertx.elasticsearch.action.bulk.BulkResponse().apply {

  if (errors != null) {
    this.setErrors(errors)
  }
  if (items != null) {
    this.setItems(items.toList())
  }
  if (status != null) {
    this.setStatus(status)
  }
  if (took != null) {
    this.setTook(took)
  }
}

